src/entities/movie.entity.ts

```diff
@@ -0,0 +1,71 @@
+import { Entity, Column, PrimaryGeneratedColumn, CreateDateColumn, UpdateDateColumn, ManyToMany, JoinTable, ManyToOne, JoinColumn, BeforeInsert, OneToMany } from 'typeorm';
+import { v4 as uuidv4 } from 'uuid';
+import { Director } from './director.entity';
+import { Actor } from './actor.entity';
+import { Festival } from './festival.entity';
+import { MovieActor } from './movie_actor.entity';
+import { MovieAward } from './movie_award.entity';
+
+
+@Entity()
+export class Movie {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @Column({ type: 'timestamptz' })
+  release_date: Date;
+
+  @Column()
+  title: string;
+
+  @Column({ nullable: true })
+  poster: string;
+
+  @Column({ type: 'float', nullable: true })
+  average_rating: number;
+
+  @CreateDateColumn({ type: 'timestamptz' })
+  created_at: Date;
+
+  @UpdateDateColumn({ type: 'timestamptz' })
+  updated_at: Date;
+
+  @Column({ unique: true, type: 'uuid' })
+  uuid: string;
+
+  @Column({ nullable: true })
+  trailer: string;
+
+  @Column({nullable:true})
+  duration:string;
+
+  @Column({nullable:true})
+  overview:string;
+
+  @Column({nullable:true})
+  language:string;
+
+  @Column("text", { array: true, default: [] })
+  genres: string[];
+
+  @ManyToOne(() => Director, director => director.movies)
+  @JoinColumn({ name: 'director_id' })
+  director: Director;
+
+  @ManyToOne(() => Director, writer => writer.movies, { nullable: true })
+  @JoinColumn({ name: 'writer_id' })
+  writer: Director;
```

Comment:
How can a writer be of type Director, u need to change `Director` entity to movie worker/maker/cast, etc.,  something generic as now it has a type that identifies the role of this cast member

Reply:
I can't fix this to be honest i tried to change it but i could not find all the imports to fix it